contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -3425,14 +3344,49 @@ private String switchLTIError(int result, String method){
 				return method + ": generic LTI error";
 		}
 	}
-	
-	private void processError(ContentReviewItem item, Long status, String error){//TODO resto de llamadas
-		item.setStatus(status);
-		if(error != null){
+
+	private void processError( ContentReviewItem item, Long status, String error, Integer errorCode )
+	{
+		if( status == null )
+		{
+			IllegalArgumentException ex = new IllegalArgumentException( "Status is null; you must supply a valid status to update when calling processError()" );
+			log.warn( ex );
```

Comment:
With the very messy lock/unlock handling I'm not sure that throwing an exception is the most robust thing todo here.

If the lock management (`obtainLock/releaseLock`) was in a `try/finally` it's much easier to be sure that you're not going to leave locked items around.